pipelines/tekton/aiedge-e2e/tasks/model-registry-download.yaml

```diff
@@ -0,0 +1,116 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: model-registry-download
+spec:
+  description: This Task can be used to fetch a model properties from model registry when enabled or properties passed by pipelinerun 
+  params:
+  - name: model-name
+    type: string
+  - name: model-version
+    type: string
+  #Values to be passed by pipeline run when model-register is disabled
+  - name: s3-bucket-name
+    type: string
+    default: ""
+  - name: modelRelativePath
+    type: string
+  - name: model-dir
+    type: string
+    default: "."
+  - name: fetch-model
+    type: string
+  - name: git-model-repo
+    type: string
+    default: ""
+  - name: git-model-revision
+    type: string
+    default: "main"
+  - name: model-registry-hostname
+    type: string
+    default: ""
+  - name: model-registry-enabled
+    type: string
+  results:
+  - name: fetch-model
+    type: string
+    description: The fetch mechanism that will be forwarded to the pipeline
+  - name: git-model-repo
+    description: The GIT URI that will be forwarded to the pipeline
+  - name: s3-bucket-name
+    description: The s3 Bucket name that will be forwarded to the pipeline
+  - name: modelRelativePath
+    type: string
+  - name: model-dir
+    type: string
+  - name: git-model-revision
+    type: string
+  steps:
+  - name: download-model-registry
+    image: registry.access.redhat.com/ubi9/ubi
+    script: |
+      #!/usr/bin/env bash
+
+      set -Eeuo pipefail
+
+      if [ "$(params.model-registry-enabled)" == "true" ]; then
+        yum install jq -y
+
+        export MODEL_ID=$(curl --silent -X 'GET' \
+        "$(params.model-registry-hostname)/api/model_registry/v1alpha3/registered_models?" -H 'accept: application/json' | \
+        jq -r --arg MODEL_NAME "$(params.model-name)" '.items[] | select(.name == $MODEL_NAME).id')
+        
+        export MODEL_VERSION_ID=$(curl -X 'GET' \
+        "$(params.model-registry-hostname)/api/model_registry/v1alpha3/registered_models/$MODEL_ID/versions?" -H 'accept: application/json' | \
+        jq -r --arg MODEL_VERSION "$(params.model-version)" '.items[] | select(.name == $MODEL_VERSION).id')
+        
+        echo -n "MODEL ID is $MODEL_ID"
+        echo -n "MODEL VERSION ID is $MODEL_VERSION_ID"
+
+        if [[ ! -z "$MODEL_ID" ]]; then
```

Comment:
Assuming that an empty `MODEL_ID` is an unrecoverable failure, we should exit with a non-zero error code to prevent any further execution of the pipeline steps since the model information is missing